### PR TITLE
Use the monotonic clock as much as possible (+ 1 minor fix)

### PR DESCRIPTION
**Pull request by fhuberts**
So we're not screwed when the clock is adjusted

Signed-off-by: Ferry Huberts <ferry.huberts@pelagic.nl>

### CODE REVIEW

**Comment by alexgrin** [comment]
Good call on CLOCK_MONOTONIC

**Comment by fhuberts** [comment]
I'd really like to use CLOCK_MONOTONIC_RAW but I'm unsure whether that is supported by all target platforms

File: service.c [hunk 1]
```diff
@@ -422,10 +429,11 @@ srv_main_loop(service_context sc)
 	/*
 	 * then wait for either inputs or the next scheduled timer to go off
 	 */
+  timespec_to_timeval(&sc->gbl_timer, &timeval);
 	if (sc->ninputs || sc->noutputs) {
```
Comment (bcopeland) [inline service.c]: Looks like a whitespace issue here, can you fix in a follow-up?

Comment (fhuberts) [inline service.c]: done, see #36 

threw in #37 as a bonus

**Pull request merged**